### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by gilbertsoft**

### CODE REVIEW

File: Classes/Custom/DataHandler.php [hunk 1]
```diff
@@ -68,7 +68,7 @@ public function processDatamap_postProcessFieldArray( $status, $table, $id, arra
 
 
 		// Only listen to own plugin data
-		if( !empty( $listType ) && compare( $listType, 'aimeos_', 1 ) === 0 )
+		if( !empty( $listType ) && substr_compare( $listType, $prefix, 1, strlen( $prefix ) ) === 0 )
```
Comment (aimeos) [inline Classes/Custom/DataHandler.php]: `prefix` seems to be undefined. Why is the first character skipped?

Comment (gilbertsoft) [inline Classes/Custom/DataHandler.php]: oops, was there, but then deleted!? fixed and an other bug too.

File: Classes/Custom/DataHandler.php [hunk 2]
```diff
@@ -67,8 +67,10 @@ public function processDatamap_postProcessFieldArray( $status, $table, $id, arra
 		}
 
 
+		$prefix = 'aimeos_';
+
 		// Only listen to own plugin data
-		if( !empty( $listType ) && compare( $listType, 'aimeos_', 1 ) === 0 )
+		if( !empty( $listType ) && substr_compare( $listType, $prefix, 0, strlen( $prefix ) ) === 0 )
```
Comment (aimeos) [inline Classes/Custom/DataHandler.php]: How does the content of `$listtype` looks like?

Comment (gilbertsoft) [inline Classes/Custom/DataHandler.php]: the plugin name, e.g. aimeos_catalog-detail

Comment (gilbertsoft) [inline Classes/Custom/DataHandler.php]: But found an other problem. Was a really bad example I have found! For the moment it does nothing, but you could use it to remove deprecated fields if the flexform xml file is delete. Should I delete the file or what do we do? The ext_localconf.php has to be cleaned to...

**Comment by gilbertsoft** [comment]
The rest seems to be ok. Did some test with plugin settings only, constant editor only and plugin settings which overrides the global settings. Did no test with tssetup but there finally nothing has changed.

File: Classes/Custom/DataHandler.php (hunk 2 above)
Comment (aimeos) [inline Classes/Custom/DataHandler.php]: `strncmp( $listType, 'aimeos_', 7 )` would be better in this case.

Doesn't it work at all? I thought we need that as soon as the eID dispatcher for the JSON calls is used?

Comment (gilbertsoft) [inline Classes/Custom/DataHandler.php]: I'm currently testing the eID changes...

**Pull request merged**